torch/csrc/utils/tensor_numpy.cpp

```diff
@@ -289,34 +292,31 @@ bool is_numpy_scalar(PyObject* obj) {
          PyArray_IsScalar(obj, Floating) || PyArray_IsScalar(obj, ComplexFloating));
 }
 
-at::Tensor tensor_from_cuda_array_interface(PyObject* obj) {
+at::Tensor tensor_from_generic_array_interface(PyObject* obj, THPObjectPtr& ai_dict, DeviceType device_type) {
   if (!is_numpy_available()) {
+    // numpy is required for parsing type descriptor string
     throw std::runtime_error("Numpy is not available");
   }
-  auto cuda_dict = THPObjectPtr(PyObject_GetAttrString(obj, "__cuda_array_interface__"));
-  TORCH_INTERNAL_ASSERT(cuda_dict);
-
-  if (!PyDict_Check(cuda_dict)) {
-    throw TypeError("`__cuda_array_interface__` must be a dict");
-  }
-
-  // Extract the `obj.__cuda_array_interface__['shape']` attribute
-  std::vector<int64_t> sizes;
-  {
-    PyObject *py_shape = PyDict_GetItemString(cuda_dict, "shape");
-    if (py_shape == nullptr) {
-      throw TypeError("attribute `shape` must exist");
+  if (!PyDict_Check(ai_dict)) {
+    // numpy compatible exception:
+    switch(device_type) {
+    case kCPU: throw ValueError("Invalid __array_interface__ value, must be a dict");
```

Comment:
Use the existing error macros like `TORCH_CHECK(false, ...)` or, in this case, the more specific `TORCH_CHECK_VALUE(false, ...)`.

See 

https://github.com/pytorch/pytorch/blob/22a34bcf4e5eaa348f0117c414c3dd760ec64b13/c10/util/Exception.h#L374